packages/web-config-server/src/apiV1/utils/findAccessibleGroupedMapOverlays.js

```diff
@@ -59,26 +61,62 @@ const findNestedGroupedMapOverlays = async (
         childMapOverlayGroupRelations,
         accessibleMapOverlays,
       );
-
-      mapOverlayGroupResults.push({
+      const mapOverlayGroupResult = integrateMapOverlayReference({
         id: mapOverlayGroupRelation.child_id,
-        name,
+        groupName: name,
         children,
       });
+      mapOverlayGroupResults.push(mapOverlayGroupResult);
     }
   }
-
   const sortedMapOverlayResults = sortMapOverlayItems(
     mapOverlayGroupResults.concat(mapOverlayResults),
     mapOverlayItemRelations,
   );
   return sortedMapOverlayResults.map(item => {
-    //id was added only for sorting purposes, remove it because it is not required in the front end
+    // id was added only for sorting purposes, remove it because it is not required in the front end
     const { id, ...itemToReturn } = item;
     return itemToReturn;
   });
 };
 
+// Only display reference info on map overlay group if all map overlays have same reference.
+const integrateMapOverlayReference = ({ id, groupName, children }) => {
```

Comment:
slightly nit picky but since `children` can contain both overlays and overlay groups, maybe it's better to change all the variables `mapOverlays`/`mapOverlay` here to `mapOverlayItems`/`mapOverlayItem`?

Reply:
Good idea Chris 👍